lib/checkmemoryleak.cpp

```diff
@@ -996,7 +996,7 @@ void CheckMemoryLeakNoVar::checkForUnreleasedInputArgument(const Scope *scope)
         const Token* tok2 = tok->next()->astParent();
         while (tok2 && tok2->isCast())
             tok2 = tok2->astParent();
-        if (tok2 && tok2->isAssignmentOp())
+        if (Token::Match(tok2, "%assign%|return"))
```

Comment:
I am not completely sure that the null pointer check can be omitted with `Token::Match`. Please correct me if it can't.

Reply:
It can be omitted, here are the relevant lines in `Token::Match()`.
https://github.com/danmar/cppcheck/blob/c9a9b56e0ef69d5a10c8a1c92b404be316e5a196/lib/token.cpp#L631-L638

Reply:
@rikardfalkeborn Thank you, I don't know why I missed that. Do you know if there is a syntax summary somewhere for `Token::match`? If I go by the code I am likely to miss details.

Reply:
It is documented pretty well in Token.h, even though it doesn't say what happens when `tok` is `NULL`. Thankfully, I don't think I've ever been forced to look at the implementation details of `Token::Match()`, there's a lot to digest there. 😀

https://github.com/danmar/cppcheck/blob/c9a9b56e0ef69d5a10c8a1c92b404be316e5a196/lib/token.h#L274-L317